nsfc-plugin/src/test/java/org/osc/controller/nsfc/AbstractNeutronSfcPluginTest.java

```diff
@@ -19,75 +19,65 @@
 import static java.util.Collections.singletonList;
 import static org.osc.controller.nsfc.TestData.*;
 
-import javax.persistence.EntityManager;
-
 import org.junit.After;
 import org.junit.Before;
 import org.junit.Rule;
 import org.junit.rules.ExpectedException;
-import org.mockito.Answers;
 import org.mockito.Mock;
+import org.mockito.Mockito;
 import org.openstack4j.api.Builders;
-import org.osc.controller.nsfc.entities.InspectionHookEntity;
-import org.osc.controller.nsfc.entities.InspectionPortEntity;
+import org.openstack4j.api.OSClient.OSClientV3;
+import org.openstack4j.api.client.IOSClientBuilder.V3;
+import org.openstack4j.api.networking.NetworkingService;
+import org.openstack4j.api.networking.PortService;
+import org.openstack4j.api.networking.ext.ServiceFunctionChainService;
 
 public abstract class AbstractNeutronSfcPluginTest {
+    @Mock
+    protected V3 v3;
+
+    @Mock
+    protected OSClientV3 osClient;
+
+    @Mock
+    protected ServiceFunctionChainService sfcService;
+
+    @Mock
+    protected NetworkingService networkingService;
 
-    @Mock(answer = Answers.CALLS_REAL_METHODS)
-    protected TestTransactionControl txControl;
+    @Mock
+    protected PortService portService;
 
-    protected EntityManager em;
 
     @Rule
     public ExpectedException exception = ExpectedException.none();
 
     @Before
     public void setup() throws Exception {
-        this.em = InMemDB.getEntityManager();
-        this.txControl.init(this.em);
         setupDataObjects();
+        Mockito.when(this.networkingService.port()).thenReturn(this.portService);
+        Mockito.when(this.sfcService.portchains()).thenReturn(portChainService);
+        Mockito.when(this.sfcService.portpairs()).thenReturn(portPairService);
+        Mockito.when(this.sfcService.portpairgroups()).thenReturn(portPairGroupService);
+        Mockito.when(this.osClient.sfc()).thenReturn(this.sfcService);
+        Mockito.when(this.osClient.networking()).thenReturn(this.networkingService);
+
     }
 
     @After
     public void tearDown() throws Exception {
-        InMemDB.close();
     }
 
-    protected InspectionHookEntity persistInspectionHook() {
+    protected void persistInspectionHook() {
```

Comment:
add a todo here?

persistInspectionHook cannot be same as  persistInspectionPort